File: leetcode/206.ReverseLinkedList/solutionV2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode backHead = head;
        ListNode cur = head.next;
        head.next = null;
        ListNode tmp = head;
        while(cur != null) {
            tmp = cur;
            cur = cur.next;
            tmp.next = backHead;
            backHead = tmp;
        }
        
        return backHead;
    }
}

// saves a ListNode.

File: leetcode/82.RemoveDuplicates2/solutionV2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if(head === null){
        return null;
    }
    if(head.next !== null && head.val === head.next.val){
        while(head.next !== null && head.val === head.next.val){
            head = head.next;
        }
        return deleteDuplicates(head.next);
    }else{
        head.next = deleteDuplicates(head.next)
    }
    return head;
};

// 有点意思， 递归。